x/upgrade/types/plan.go

```diff
@@ -45,3 +47,25 @@ func (p Plan) ShouldExecute(ctx sdk.Context) bool {
 func (p Plan) DueAt() string {
 	return fmt.Sprintf("height: %d", p.Height)
 }
+
+// UpgradeDetails is a struct that represents the details of an upgrade
+// This is held in the Info object of an upgrade Plan
+type UpgradeDetails struct {
```

Comment:
is upgrade info used elsewhere? is it possible theres other fields also passed in that might mess up the parsing? maybe we add a test cast like:
```
{"upgradeType": "minor", "otherInfo": "foo"}
```

Reply:
`encoding/json` unmarshaling is tolerant to extra & missing fields (fwiw).  The Info object is used by cosmosvisor if we choose to support auto-download (which would include a cpu-arch->binary map), but otherwise not used.